router/src/server.rs

```diff
@@ -31,6 +33,10 @@ use tracing::{info_span, instrument, Instrument};
 use utoipa::OpenApi;
 use utoipa_swagger_ui::SwaggerUi;
 
+lazy_static! {
+    static ref MODEL_ID: Mutex<String> = Mutex::new("".to_string());
```

Comment:
What does this do?

Reply:
It smells like we're setting a global lock on the current model ID, which would break concurrency?

Reply:
It allows you to create a global var that we then update [here](https://github.com/predibase/lorax/pull/220/files#diff-58724fd9e4e3f359bc071bec55c6b3f8149372efa3dc33981ada5bf60afee878R844-R847) with the model name. Needed so it is not read only.

Reply:
But why do we need it? I see we set it, but don't see where it gets used.

Reply:
I see, but why do we need to lock every time we read it if it's only initialized once at the beginning and then read-only? Feels like we're taking a concurrency hit for no reason.

Reply:
@tgaddair we just set it once when we run the server for the first time. Maybe we can have a read-write lock. Or maybe we can do something different but we want to communicate the value of model_id [here](https://github.com/predibase/lorax/pull/220/files#diff-58724fd9e4e3f359bc071bec55c6b3f8149372efa3dc33981ada5bf60afee878R768) to the request handler [here](https://github.com/predibase/lorax/pull/220/files#diff-58724fd9e4e3f359bc071bec55c6b3f8149372efa3dc33981ada5bf60afee878R855). Maybe we can create a partial function like in python? idk

Reply:
@tgaddair I think we would need to lock a global var but I could be wrong here

Reply:
Actually looks like I can just use this https://docs.rs/once_cell/latest/once_cell/

Reply:
Looks like exactly what we want! Nice find.